packages/x-date-pickers/src/internals/hooks/usePicker/usePickerValue.types.ts

```diff
@@ -301,8 +301,8 @@ export interface UsePickerValueActions {
   onDismiss: () => void;
   onCancel: () => void;
   onSetToday: () => void;
-  onOpen: () => void;
-  onClose: () => void;
+  onOpen: (event: React.UIEvent) => void;
+  onClose: (event?: React.UIEvent) => void;
```

Comment:
Do we have places where we pass the event to `onClose`?
This PR only passes them to `onOpen`

Reply:
Explicitly nope, but implicitly it is called by clicking on the open picker button and this is basically the reason why our selection indexes get screwed up... 🙈 
Without this, the fix doesn't work as the input selection logic gets called.

Reply:
OK make sense :+1: